proxy/src/main.rs

```diff
@@ -54,42 +57,30 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     };
 
     let temp_dir = tempfile::tempdir().expect("test dir creation failed");
-    let tmp_path = temp_dir.path().to_str().expect("path extraction failed");
 
-    let mut peer = if args.test {
-        let key = keys::SecretKey::new();
-        let config = coco::default_config(key, tmp_path).expect("failed to get config");
-        coco::Peer::new(config)
-            .await
-            .expect("failed to create /tmp user peer")
+    let (pw, path) = if args.test {
+        let pw = SecUtf8::from("asdf");
+        (pw, Some(temp_dir.path().to_path_buf()))
     } else {
-        // TODO(finto): There should be a coco::config module that knows how to parse the
-        // configs/parameters to give us back a `PeerConfig`
-
-        // TODO(finto): Should be fetched from key storage
-        let key = keys::SecretKey::new();
-        // TODO(finto): Should be read from config file
-        let gossip_params = Default::default();
-        // TODO(finto): Read from config or passed as param
-        let listen_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 0);
-        // TODO(finto): could we initialise with known seeds from a cache?
-        let seeds: Vec<(peer::PeerId, SocketAddr)> = vec![];
-        let disco = discovery::Static::new(seeds);
-        // TODO(finto): read in from config or passed as param
-        let paths = paths::Paths::new()?;
-        let config = net::peer::PeerConfig {
-            key,
-            paths,
-            listen_addr,
-            gossip_params,
-            disco,
-        };
+        let pw = pinentry::Prompt::new("please provide your Radicle passphrase: ")
```

Comment:
Does that mean the startup is currently halted until the user entered a passphrase?

Reply:
Yeah... this is not ideal. There isn't a secure way to supply the passphrase to `main`, however: if you pass it as a CLI argument or read it from an environment variable, everyone can see it.

For now, I would acknowledge that the `proxy` is not run as a standalone daemon, and simply start it up in "locked mode". That is, it needs to be "unlocked" via an API request. As a second step, we can think about some kind of vault interface, enabling secure standalone operations.

Reply:
I'm not sure I understand this "locked" and "unlocked" API request business :thinking:

You need key storage to be able to configure and create a Peer. And the Peer gets propagated to all the API handlers. So I'm not sure, personally, what this would like.

Reply:
I'd suggest we use a fake pw for now as we do for `args.test` and lock at that password problem as follow-up.

Reply:
Nothing which requires access to the key works until the `proxy` process is able to decrypt the key, right? Storing the passphrase somewhere where `proxy` can read it non-interactively is a no-go, because then you could store they unencrypted in the first place. Interactively requiring a passphrase to start up `proxy` is also not ideal, because, do you actually have a terminal?

So, either you have a `Pinentry` impl which _somehow_ shows a dialog box to the user (who started `upstream` by clicking on an icon) _before_ all of the backend and frontend is loaded. Or, you start `proxy` with all state `None` which would require a key to be initialised (and handle that in the API), and provide an endpoint to which the frontend can submit the passphrase, which in turn loads and decrypts the key and initialises all state which depends on it.

Reply:
> So, either you have a `Pinentry` impl which _somehow_ shows a dialog box to the user (who started `upstream` by clicking on an icon) _before_ all of the backend and frontend is loaded. Or, you start `proxy` with all state `None` which would require a key to be initialised (and handle that in the API), and provide an endpoint to which the frontend can submit the passphrase, which in turn loads and decrypts the key and initialises all state which depends on it.

As I pointed out this is something we need to align with @radicle-dev/product to see where and when we want the prompt to happen. We already exercise replacing the `Peer` and other resources we need to full-fill requests for testing purposes, so a similar strategy work there as well. A question in that context do you need the key for all coco operations or only for mutations/writes?

Reply:
> I'd suggest we use a fake pw for now as we do for args.test and lock at that password problem as follow-up.

I already have that in place. It's just when you run it without `--test` that you provide a password.

===============

> A question in that context do you need the key for all coco operations or only for mutations/writes?

Not currently because the `PeerApi` allows you to get a `SecretKey`, but that means the `SecretKey` lives in memory within the `PeerApi`. That's no bueno for :tinfoilhat: reasons. So a follow-up to this would be looking at removing the `SecretKey` from `PeerApi`.

===============

But alas! Judging from this conversation, I suppose I should remove this, create a follow-up ticket, and put in a fake password for now?

Reply:
> But alas! Judging from this conversation, I suppose I should remove this, create a follow-up ticket, and put in a fake password for now?

Yeah, to avoid this blocking this changeset and have a focused exploration around pw handling in the app.